test/Confluent.Kafka.UnitTests/Serialization/String.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright 2016-2017 Confluent Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+using Xunit;
+using System.Text;
+
+
+namespace Confluent.Kafka.Serialization.Tests
+{
+    public class StringTests
+    {
+        [Fact]
+        public void SerializeDeserialize()
+        {
+            Assert.Equal("hello world", new StringDeserializer(Encoding.UTF8).Deserialize(new StringSerializer(Encoding.UTF8).Serialize("hello world")));
+            Assert.Equal("ឆ្មាត្រូវបានហែលទឹក", new StringDeserializer(Encoding.UTF8).Deserialize(new StringSerializer(Encoding.UTF8).Serialize("ឆ្មាត្រូវបានហែលទឹក")));
+            Assert.Equal("вы не банан", new StringDeserializer(Encoding.UTF8).Deserialize(new StringSerializer(Encoding.UTF8).Serialize("вы не банан")));
+
+            Assert.NotEqual("ឆ្មាត្រូវបានហែលទឹក", new StringDeserializer(Encoding.UTF8).Deserialize(new StringSerializer(Encoding.UTF32).Serialize("ឆ្មាត្រូវបានហែលទឹក")));
```

Comment:
i was (errantly) assuming Deserialize never threw. Will remove the error case tests and replace with a TODO. Agree it's worth having these tests but it'll take a bit of time to construct them well and ROI is low i think.